llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2048,19 +2048,23 @@ class VPBlendRecipe : public VPSingleDefRecipe {
 
   VP_CLASSOF_IMPL(VPDef::VPBlendSC)
 
+  /// A normalized blend is one that has an odd number of operands, whereby the
+  /// first operand does not have an associated mask.
+  bool isNormalized() const { return getNumOperands() % 2; }
+
   /// Return the number of incoming values, taking into account that the first
   /// incoming value has no mask.
   unsigned getNumIncomingValues() const { return (getNumOperands() + 1) / 2; }
```

Comment:
This looks incorrect when the recipe is not normalised. Should it be

```
  unsigned getNumIncomingValues() const { return (getNumOperands() + isNormalized()) / 2; }
```

?

Reply:
The extra 1 would disappear during integer division.

Reply:
Sure, but it still looks a little odd. If the code is relying upon that behaviour then I think it's at least worth a comment somewhere.

Reply:
Not sure we need to add comments explaining how integer maths works.  That said, I will update the comment because it is out-of-date based on my refactoring.

Reply:
Well, it's not about how integer maths works but just future-proofing the code that's all. For example, if a future patch changes the definition of normalisation to be something else then it's less obvious to the user that this code also needs updating that's all.

Reply:
Fair enough.  I've changed to the implementation and comment to aid clarity.